UnityProject/Assets/Scripts/Systems/Explosions/Explosion.cs

```diff
@@ -48,7 +46,38 @@ public static Vector2 GetXYDirection(float angle, float magnitude)
 		public float ExplosionStrength = 0;
 		private bool InitialStep = true;
 
-		public void SetUp(int X0, int Y0, int X1, int Y1, float InExplosionStrength, bool isEmPulse)
+		private ExplosionNode NodeType;
+
+		private bool IsAnyMatchingType(ExplosionNode[] expNodes, ExplosionNode nodeType)
+		{
+			foreach(ExplosionNode expNode in expNodes)
```

Comment:
Instead of doing a loop you can just do 
expNodes[nodeType]

Reply:
Again, dont want to use enums or other hardcoded trash for that